common/headtracker/head_tracker.go

```diff
@@ -140,6 +123,45 @@ func (ht *HeadTracker[HTH, S, ID, BLOCK_HASH]) Start(ctx context.Context) error
 	})
 }
 
+func (ht *HeadTracker[HTH, S, ID, BLOCK_HASH]) handleInitialHead(ctx context.Context) error {
+	initialHead, err := ht.client.HeadByNumber(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to fetch initial head: %w", err)
+	}
+
+	if !initialHead.IsValid() {
+		ht.log.Warnw("Got nil initial head", "head", initialHead)
```

Comment:
This logic was introduced before my changes (I've just moved code around to make it clear). My assumption is that initialHead is not required to be nil to be invalid. So it might be useful to log it for debugging